improver_tests/precipitation_type/shower_condition/test_ShowerConditionProbability.py

```diff
@@ -0,0 +1,264 @@
+                        mask=np.array([[0, 0], [0, 1]]),
+                    ),
+                    "realizations": [0],
+                },
+            },
+            # Other plugin kwargs
+            {},
+            # Expected result
+            (np.array([[1, 1], [1, 0]]).astype(FLOAT_DTYPE), EXPECTED_ATTRIBUTES),
+        ),
+        # Multi-realization case with a range of probabilities returned due
+        # to variable cloud.
+        (
+            {
+                "low_and_medium_type_cloud_area_fraction": {
+                    "data": np.array(
+                        [[[0.4, 0.6], [0.4, 0.6]], [[0.6, 0.6], [0.6, 0.6]]]
+                    ).astype(FLOAT_DTYPE),
+                    "realizations": [0, 1],
+                },
+                "convective_ratio": {
+                    "data": np.zeros((2, 2, 2)).astype(FLOAT_DTYPE),
+                    "realizations": [0, 1],
+                },
+            },
+            # Other plugin kwargs
+            {},
+            # Expected result
+            (np.array([[0.5, 0], [0.5, 0]]).astype(FLOAT_DTYPE), EXPECTED_ATTRIBUTES),
+        ),
+        # Same as above, but with different threshold values applied.
+        # Cloud =< 0.7, which will result in probabilities all equal to 1.
+        (
+            {
+                "low_and_medium_type_cloud_area_fraction": {
+                    "data": np.array(
+                        [[[0.4, 0.6], [0.4, 0.6]], [[0.6, 0.6], [0.6, 0.6]]]
+                    ).astype(FLOAT_DTYPE),
+                    "realizations": [0, 1],
+                },
+                "convective_ratio": {
+                    "data": np.zeros((2, 2, 2)).astype(FLOAT_DTYPE),
+                    "realizations": [0, 1],
+                },
+            },
+            # Other plugin kwargs
+            {"cloud_threshold": 0.7},
+            # Expected result
+            (np.ones((2, 2)).astype(FLOAT_DTYPE), EXPECTED_ATTRIBUTES),
+        ),
+        # Multi-realization case with cloud and convection both providing a
+        # showery probability of 1.
+        (
+            {
+                "low_and_medium_type_cloud_area_fraction": {
+                    "data": np.array([[[0, 1], [1, 1]], [[0, 1], [1, 1]]]).astype(
+                        FLOAT_DTYPE
+                    ),
+                    "realizations": [0, 1],
+                },
+                "convective_ratio": {
+                    "data": np.array([[[0, 0], [0, 1]], [[0, 0], [0, 1]]]).astype(
+                        FLOAT_DTYPE
+                    ),
+                    "realizations": [0, 1],
+                },
+            },
+            # Other plugin kwargs
+            {},
+            # Expected result
+            (np.array([[1, 0], [0, 1]]).astype(FLOAT_DTYPE), EXPECTED_ATTRIBUTES),
+        ),
+    ),
+)
+def test_scenarios(test_cubes, kwargs, expected):
+    """Test output type and metadata"""
+    result = ShowerConditionProbability(**kwargs)(test_cubes)
```

Comment:
I always get nervous about tests where the truth condition could be modified.
```suggestion
    result = ShowerConditionProbability(**kwargs)(test_cubes.copy())
```

Reply:
Added.

Reply:
Scratch that. It's a cubelist, so adding .copy() does not work in quite the right way. The only truth used is the shape, so I've extracted that before calling the plugin.